website/docs/extensions/write/index.md

```diff
@@ -0,0 +1,134 @@
+---
+sidebar_position: 1
+title: Writing
+description: Writing my first extension
+tags: [podman-desktop, extension, writing]
+keywords: [podman desktop, extension, writing]
+---
+
+# Writing your first extension
+
+Podman Desktop extensions rely on `JavaScript` or `TypeScript`.
+
+Extensions are self-contained, all runtime dependencies are inside the final binary.
+
+#### Scaffold the project
+
+1. setup a Node.js project with `TypeScript`. Use for example `npm init` command.
+
+1. Bring the development dependency to Podman Desktop API. It allows to import the API and register the contributions.
+
+  Note that it is a dependency that brings only the definition of the API. It will not fetch any Podman Desktop internals.
+
+  ```json
+    "devDependencies": {
+      "@podman-desktop/api": "latest",
+    }
+  ```
+
+1. Setup the required metadata for the extension
+
+  ```json
+  {
+    "name": "my-extension",
+    "displayName": "My Hello World extension",
+    "description": "How to write my first extension",
+    "version": "0.0.1",
+    "icon": "icon.png",
+    "publisher": "benoitf",
+    "engines": {
+      "podman-desktop": "latest"
+    }
+  }
+  ```
+
+  In engine section, it describes the version of Podman Desktop that runs this extension.
+
+1. Includes the main entry point of the extension
+
+  ```json
+    "main": "./dist/extension.js",
+  ```
+
+1. Add Hello World command contribution
+
+  ```json
+    "contributes": {
+      "commands": [
+        {
+          "command": "my.first.command",
+          "title": "My First Extension: Hello World"
+        }
+      ]
+    },
+  ```
+
+#### Writing the codebase of the extension
+
+package.json describes the metadata of the extension. Extension activates by launching the `activate` function of the main file.
+
+##### Entry point
+Entrypoint of the extension (`main` entry in `package.json` file) needs to expose two functions.
+
+One is for activation (`activate`), one for deactivation (`deactivate`). This one is optional.
```

Comment:
```suggestion
1. (Mandatory) `activate`: activation function for the plugin.

2. (Optional) `deactivate`: deactivation function.
```

Reply:
I'm wondering why you're specifying 1. and 2. while using 1. all the time, it's computing for us the increment ?

Reply:
Both notations are correct. I believe I have been not consciously following Obsidian and IntelliJ tendency to put the actual numbering in the file, while editing in GH UI.